test/sqllogictest/transform/table_elimination.slt

```diff
@@ -0,0 +1,19 @@
+# Copyright Materialize, Inc. All rights reserved.
```

Comment:
The optimization is very zealous in this case:

```
materialize=> explain  select * from t1 union all select * from t1 except all select * from t1;;
 Optimized Plan 
----------------
 %0 =          +
 | Constant    +
```

This should have returned the rows from ```t1``` but instead returns no rows.

Reply:
http://sqlfiddle.com/#!17/b4294/1

sqlfiddle agrees and so do the result obtained with the main branch.

```
materialize=> select * from t1 union all select * from t1 except all select * from t1;
 c1 
----
  1
  2
  3
(3 rows)

materialize=> 
```

Thanks for letting me know that I can trigger the rewrite by using UNION ALL/EXCEPT ALL!